src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,130 @@
+            String line;
+            while ((line = csvReader.readLine()) != null) {
+                Task task;
+                String[] taskStr = line.split(",");
+                task = storageLineParser(taskStr);
+                tasks.add(task);
+            }
+        } catch (IOException e) {
+            System.err.println("Read failed. \nDatabase connection error.");
+        }
+        return tasks;
+    }
+
+    /**
+     * Update the {@code tasks} to the database.
+     * @param tasks tasks to update the database
+     */
+    public void update(List<Task> tasks) {
+        try {
+            File databaseFile = new File(databasePath);
+            FileWriter csvWriter = new FileWriter(databaseFile);
+            for (Task task: tasks) {
+                String[] formatTask = formatTask(task);
+                csvWriter.append(String.join(",", formatTask));
+                csvWriter.append('\n');
+            }
+            csvWriter.flush();
+            csvWriter.close();
+        } catch (IOException e) {
+            System.err.println("Creation failed. \nDatabase connection error.");
+        }
+    }
+
+    private String[] formatTask(Task task) {
+        String[] formatTask = new String[] {
+                task.getTaskType(),
+                task.getStatus() ? "1" : "0",
+                task.getTask(),
+                task.getTime().isEmpty() ? "" : task.getTime().get()
+        };
+        return formatTask;
+    }
+
+    /**
+     * Query for a specific task in the database with a query key.
+     * @param queryKey the key used to query the database
+     * @return
+     */
+    public List<Task> query(String queryKey) {
+        List<Task> tasks = new ArrayList<>();
+        try {
+            File databaseFile = new File(databasePath);
+            BufferedReader csvReader = new BufferedReader(new FileReader(databaseFile));
+            String line;
+            while ((line = csvReader.readLine()) != null) {
+                Task task;
+                if (line.contains(queryKey)) {
+                    String[] taskStr = line.split(",");
+                    task = storageLineParser(taskStr);
+                    tasks.add(task);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Read failed. \nDatabase connection error.");
+        }
+        return tasks;
+    }
+
+    private Task storageLineParser(String[] taskStr) {
+        Task task;
+        switch (taskStr[0]) {
+        case "T": {
+            task = new ToDo(taskStr[2], taskStr[1]);
+            break;
+        }
+        case "E": {
+            task = new Event(taskStr[1], taskStr[2], taskStr[3]);
+            break;
+        }
+        case "D": {
+            task = new Deadline(taskStr[1], taskStr[2], taskStr[3]);
+            break;
+        }
+        default: {
+            task = null;
+            break;
+        }
+        }
```

Comment:
those curly braces are unnecessary! i think the code would look cleaner without them~